# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserHost.cpp
# Commit: 07c8392cb04d
# Full Hash: 07c8392cb04d095caa8e65ee1a10ccc5f7352eb6
# Author: Andrew McCreight <continuation@gmail.com>
# Date: 2021-05-07 21:46:25
# Regressor Bug: 1618547
# File Overlap Count: 1
# Description:
#   Bug 1618547 - Support Fission in the process priority manager. r=kmag,gsvelto
#   
#   With Fission, there can be multiple BrowserParents in a single tab, so
#   this patch moves the tracking of active tabs onto the top BrowsingContext
#   in a tab. If the priority of a top BC is changed, then the activity
# ==============================================================================

diff -r 285861b32934 -r 07c8392cb04d dom/ipc/BrowserHost.cpp
--- a/dom/ipc/BrowserHost.cpp	Fri May 07 17:42:21 2021 +0000
+++ b/dom/ipc/BrowserHost.cpp	Fri May 07 17:51:34 2021 +0000
@@ -10,6 +10,7 @@
 #include "mozilla/dom/CancelContentJSOptionsBinding.h"
 #include "mozilla/dom/ContentParent.h"
 #include "mozilla/dom/WindowGlobalParent.h"
+#include "mozilla/ProcessPriorityManager.h"
 
 #include "nsIObserverService.h"
 
@@ -116,6 +117,8 @@
   if (!mRoot) {
     return NS_OK;
   }
+  ProcessPriorityManager::ActivityChanged(GetBrowsingContext()->Canonical(),
+                                          aRenderLayers);
   mRoot->SetRenderLayers(aRenderLayers);
   return NS_OK;
 }
@@ -149,8 +152,8 @@
   if (!mRoot) {
     return NS_OK;
   }
-  VisitAll(
-      [](BrowserParent* aBrowserParent) { aBrowserParent->Deprioritize(); });
+  ProcessPriorityManager::ActivityChanged(GetBrowsingContext()->Canonical(),
+                                          /* aIsActive = */ false);
   return NS_OK;
 }
 